### DISCUSSION TITLE
suitable MIDI Files

### DISCUSSION THREAD

**Discussion by Peer256** [onlaj/Piano-LED-Visualizer#275] [Q&A]
Hello everyone,

we managed to get the system working and it looks great! THX to @onlaj and all the contributors. Now we are struggeling with suitable MIDI Files. After downloading various MIDI files from different sources we cannot figure out why some of them are working just fine in listening mode while others fail to load with an Error in the little loading screen on the LCD. 

Is there an instruction tutorial we missed in the repository? 
Can anyone recommend a source which provides Piano MIDI Files that work?

We looked into the MIDI Files with an MidiEditor from Markus Schwenk (midieditor.org) and could not figure out why some Files work and some just don't. 2 Tracks for left and right hand titled Track 0 and Track 1 are present in both files. 

Thanks for your help!

kind regard,

Peer

**Comment by onlaj** [comment]
Hello.
I think this question should be directed to [mido](https://github.com/mido/mido)'s authors, I'm just using their "[Iterating Over Messages](https://mido.readthedocs.io/en/latest/midi_files.html#iterating-over-messages)" implementation with [some changes](https://github.com/onlaj/Piano-LED-Visualizer/blob/master/lib/functions.py#L30) to avoid drift.
